lib/SymbolGraphGen/JSON.cpp

```diff
@@ -105,6 +106,16 @@ void swift::symbolgraphgen::serialize(const Requirement &Req,
     OS.attribute("kind", Kind);
     OS.attribute("lhs", Req.getFirstType()->getString());
     OS.attribute("rhs", Req.getSecondType()->getString());
+    
+    // If the RHS type has a USR we can link to, add it to the output
+    if (!Req.getSecondType()->isTypeParameter() && !Req.getSecondType()->hasArchetype()) {
```

Comment:
Maybe you want Req->getSecondType()->hasAnyNominal() here? The predicate as written will also return true for function types that don't contain generic parameters, metatypes, etc.

Reply:
`Array<T>` will return false from `isTypeParameter` because that function is meant to check for the `T` itself - there's a separate `hasTypeParameter` for checking whether a type *has* parameters, as afar as i can tell. The separate `hasArchetype` call is meant to guard against an assertion in `printTypeUSR`.

`getAnyNominal` does seem like a better choice for what this needs, though - i'll take a look at that instead.